File: nba-scala/nba-scala-scraper/src/main/scala/com/faeddalberto/nbastats/domain/Conference.scala
package com.faeddalberto.nbastats.domain

import com.faeddalberto.nbastats.domain.Division._

object Conference {

  sealed trait Conference{val divisions :Set[Division]}

  case object EASTERN extends Conference {
    override val divisions = Set(Atlantic, Central, Southeast)
  }

  case object WESTERN extends Conference {
    override val divisions = Set(Pacific, Northwest, Soutwest)
  }

  val elements = Set(EASTERN, WESTERN)

  def apply(division :Division) :Conference = {

    if (EASTERN.divisions.contains(division))
      EASTERN
    else if (WESTERN.divisions.contains(division))
      WESTERN
    else
      throw new IllegalArgumentException("Cannot identify Conference for division " + division)

  }
}

File: nba/nba-model/src/main/java/com/faeddalberto/nbastats/model/enums/Conference.java
package com.faeddalberto.nbastats.model.enums;

public enum Conference {

    EASTERN_CONFERENCE("Eastern Conference"),
    WESTERN_CONFERENCE("Western Conference");

    Conference(String value) {
        this.value = value;
    }

    private String value;

    public String getValue() {
        return value;
    }

    public static Conference fromString(String text) {
        if (text != null) {
            for (Conference conference : Conference.values()) {
                if (text.equalsIgnoreCase(conference.value)) {
                    return conference;
                }
            }
        }
        throw new IllegalArgumentException("Cannot map " + text + " to a valid conference");
    }
}
